tensorflow/python/ops/math_ops.py

```diff
@@ -438,9 +439,13 @@ def divide(x, y, name=None):
     # override names. Use a dummy class to track the runtime division behavior
     return DivideDelegateWithName(x, name) / y
   else:
+    if not (isinstance(x, ops.Tensor)  or isinstance(y, ops.Tensor)):
+      if sys.version_info.major < 3:
+        return _truediv_python2(x, y)
+      else:
+        return _truediv_python3(x, y)
     return x / y
```

Comment:
I'm sorry, I may be missing something here, but is the logic above saying that we'll just use `_truediv_python` for non-tensor inputs? If so, then the test case shouldn't pass, since we should get a python value back.

I'm probably misreading the code. If so, please document or restructure to help clarify.

Reply:
Okay I think I understand this PR better, I was misreading the if condition.

Wouldn't it be simpler to simply call convert to tensor on `x` and then let the operator overloads kick in? That's basically what each of those function do in the first place.

Reply:
Please remove the comment or simply to simple say we do the convert in order always return a Tensor. Also, I think the code should be:
```
tensor_util.is_tensor(x):
  x = ops.convert_to_tensor(x)
```

Reply:
Why do you need to check `tensor_util.is_tensor(y)`?

Reply:
@jaingaurav I saw the logic in `_OverrideBinaryOperatorHelper` will try both `__truediv__` and rtruediv__` (LHS or RHS). I was overly concerned about if convert x may cause issues when y is a different dtype. So initially tried to test both.

But now I give it try with always convert x. Looks like all tests pass so it should be fine. Thanks for the help!